src/realm/object-store/object_schema.cpp

```diff
@@ -49,13 +49,14 @@ ObjectSchema::ObjectSchema(std::string name, std::initializer_list<Property> per
 {
 }
 
-ObjectSchema::ObjectSchema(std::string name, IsEmbedded is_embedded,
+ObjectSchema::ObjectSchema(std::string name, IsEmbedded is_embedded, IsAsymmetric is_asymmetric,
```

Comment:
you should not change the existing constructor, but define the two new ones declared in the header file.

Reply:
Can an embedded schema be asymmetric?

Reply:
No. An embedded table cannot be asymmetric. Not sure what that would mean

Reply:
An embedded table is implicitly asymmetric if its top-level parent is asymmetric. I don't think we need to require people to explicitly state that in their schema, but I don't think it's necessarily an error either.

Reply:
But an embedded table can be linked to from multiple top-level tables - both symmetric and asymmetric. If we want to support that scenario, then we should probably not annotate the embedded table for symmetry.

Reply:
That is true / a good point. The embedded table can be a part of multiple different tables. For what its worth, this is why the server did this by using an enum where a table is either: 
1. TopLevel 
2. Embedded 
3. Asymetric 

In this case, Asymmetric means a top-level table that has been deemed Asymmetric. No Embedded tables are annotated as Asymmetric since it doesnt matter for us (and im not sure if it needs to matter for you)

Reply:
do we have a consensus on this?

Reply:
Its up to you. We can either ensure that it is an error to be both asymmetric and embedded or we can change the method signature to take an enum instead of two bools, but that is up to you 🤷‍♂️

Reply:
I would prefer having an enum so we don't have to perform checks all over the place..